lib/Dialect/FIRRTL/Transforms/InjectDUTHierarchy.cpp

```diff
@@ -28,6 +31,42 @@ struct InjectDUTHierarchy : public InjectDUTHierarchyBase<InjectDUTHierarchy> {
 };
 } // namespace
 
+/// Add an extra level of hierarchy to a hierarchical path that places the
+/// wrapper instance after the DUT.  E.g., this is converting:
+///
+///   firrtl.hierpath [@Top::@dut, @DUT]
+///
+/// Int:
+///
+///   firrtl.hierpath [@Top::@dut, @DUT::@wrapper, @Wrapper]
+static void addHierarchy(HierPathOp path, FModuleOp dut,
+                         InstanceOp wrapperInst) {
+  auto namepath = path.namepath().getValue();
+
+  size_t nlaIdx = 0;
+  while (path.modPart(nlaIdx) != dut.getNameAttr())
+    ++nlaIdx;
+
+  // Add everything up to the DUT.
+  auto front = namepath.take_front(nlaIdx);
+  SmallVector<Attribute> newNamepath(front.begin(), front.end());
```

Comment:
```c++
  SmallVector<Attribute> newNamepath;
  newNamepath.reserve(namepath.size() + 1);
  while (path.modPart(nlaIdx) != dut.getNameAttr())
    newNamepath.push_back(namepath[nlaIdx++]);
```

Reply:
Switching to this in a `!fixup` commit I'll push shortly.